cas-server-support-jdbc/src/main/java/org/jasig/cas/services/JpaServiceRegistryDaoImpl.java

```diff
@@ -35,6 +37,7 @@
     @PersistenceContext
     private EntityManager entityManager;
 
+    @Transactional(readOnly = false)
```

Comment:
Reviewed. Here is what we have on CASImpl:

1. destroyTicketGrantingTicket: single call to the registry to destroy.
2. grantServiceTicket: single call to the registry to add ticket
3. delegateTicketGrantingTicket: single call to the registry to add ticket
4. validateServiceTicket: single call to the registry to delete ticket
5. getTickets: single call to the registry to get ticket
6. createTicketGrantingTicket: single call to the registry to add ticket

As you see, we always have single transactions.